rfcs/0022-tir-non-scalar-constants.md

```diff
@@ -0,0 +1,107 @@
+
+- Feature Name: tir_non_scalar_constants
+- Start Date: 2021-06-01
+- RFC PR: https://github.com/apache/tvm-rfcs/pull/22
+- GitHub Issue: TBD
+
+# 1. Summary
+
+This RFC proposes how non-scalar constants could be represented in TIR and used by passes in the lowering process.
+
+# 2. Motivation 
+
+Currently, the non-scalar constants could be represented in Relay (relay.Constant) to be used by relay passes but not in TIR. Therefore, when performing lowering using TIR passes, we have to maintain a side-channel of tir::Var to constant non-scalar data mapping to perform transformations that could use the knowledge where some of the data are constants.
```

Comment:
flip the second sentence around: "Therefore, when a compiler pass is operating on TIR, it needs additional input to determine whether a `tir::Var` is runtime-supplied data or a compile-time constant (and what that constant data is). Should that pass want to transform the constant data, the additional input also becomes an output. In order to provide this data, the compiler has to maintain a side-channel map, complicating the Pass interface."